src/components/application_manager/src/application_data_impl.cc

```diff
@@ -555,26 +555,18 @@ void DynamicApplicationDataImpl::set_display_capabilities(
       return false;
     }
 
-    const auto find_res = std::find_if(
-        tmp_window_capabilities_arr->begin(),
-        tmp_window_capabilities_arr->end(),
-        [&window_id](const smart_objects::SmartObject& element) {
-          LOG4CXX_DEBUG(logger_,
-                        "Searching for "
-                            << window_id << " with element window id "
-                            << element[strings::window_id].asInt());
-          if ((!element.keyExists(strings::window_id)) && 0 == window_id) {
-            return true;
-          }
-          const auto current_window_id = element[strings::window_id].asInt();
-          if (window_id == current_window_id) {
-            return true;
-          }
-          return false;
-        });
-
-    return find_res != tmp_window_capabilities_arr->end();
+    for (auto element : *tmp_window_capabilities_arr) {
+      if (element.keyExists(strings::window_id)) {
+        if (window_id == element[strings::window_id].asInt())
+          return true;
+      } else if (window_id == 0) {
+        return true;
+      }
```

Comment:
@iCollin May be this way be better? 
```cpp
if (element.keyExists(strings::window_id)) {
  if (helpers::Compare<int, EQ, ONE>(window_id, 0, element[strings::window_id].asInt())) {
  return true;
}
```

Reply:
`window_id` 0 matches only if the `window_id` key does not exist on the element.

Reply:
@iCollin I see. Thank you. I missed curly braces...